### PR TITLE
Implement the tmt run --follow option

### PR DESCRIPTION
**Pull request by FrNecas**
Something up for discussion: do we want to exactly mimic `tail -f` behaviour or do we want to somehow modify it? Tail shows the last 10 lines in the file and then adds more once they are added. I followed this approach as I think it's pointless to show the whole log but it may be useful to show at least a few of the last lines to give context.

Resolves: #482

### CODE REVIEW

File: tmt/base.py [hunk 1]
```diff
@@ -821,6 +826,10 @@ def __init__(self, id_=None, tree=None, context=None):
             if id_ is None:
                 raise tmt.utils.GeneralError(
                     "No last run id found. Have you executed any run?")
+        id_specified = context.params.get('last') or context.params.get('id')
+        if context.params.get('follow') and not id_specified:
```
Comment (psss) [inline tmt/base.py:830]: Specifying `--id` ends up with an error. Let's simplify the check a bit:

```python
if context.params.get('follow') and id_ is None:
```

And remove the preceding line altogether.

File: tmt/base.py [hunk 2]
```diff
@@ -942,13 +951,39 @@ def finish(self):
             raise SystemExit(0)
         raise SystemExit(2)
 
+    def follow(self):
```
Comment (psss) [inline tmt/base.py:954]: Please, include a docstring.

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for implementing this. Looks fine. Added just a couple of minor suggestions.

File: tmt/cli.py [hunk 3]
```diff
@@ -188,6 +188,9 @@ def main(click_contex, root, context, **kwargs):
 @click.option(
     '-r', '--rm', '--remove', 'remove', is_flag=True,
     help='Remove the workdir when test run is finished.')
+@click.option(
+    '--follow', is_flag=True,
+    help='Output the logfile as it grows.')
```
[the remark this answers is not in the document]
Comment (FrNecas) [reply tmt/cli.py:193]: I think we could but wouldn't it be a bit confusing adding `-F` to run when -F is already used in some steps (for example flavor in minute or --filter in discover --how fmf)?

[the remark this answers is not in the document]
Comment (psss) [reply tmt/cli.py:193]: Well, I expect there will be more such collisions and I don't see a big problem in that. On the other hand, we can wait a bit with the short option until other new features are implemented (which could bring another candidate for the `-F` shortcut). I'm ok with both.

[the remark this answers is not in the document]
Comment (FrNecas) [reply tmt/cli.py:193]: Ok, lets stay with the long option only for now; we can add the short one later if the need arises and we don't find a better use for `-F`.

**Review by psss — APPROVED** [review]
Thanks for addressing all issues. Looks good.

**Pull request merged**